app/src/test/java/org/simple/clinic/selectstate/SelectStateUpdateTest.kt

```diff
@@ -0,0 +1,58 @@
+package org.simple.clinic.selectstate
+
+import com.spotify.mobius.test.NextMatchers.hasModel
+import com.spotify.mobius.test.NextMatchers.hasNoEffects
+import com.spotify.mobius.test.UpdateSpec
+import com.spotify.mobius.test.UpdateSpec.assertThatNext
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.ResponseBody.Companion.toResponseBody
+import org.junit.Test
+import org.simple.clinic.TestData
+import org.simple.clinic.appconfig.StatesResult.FetchError
+import org.simple.clinic.appconfig.StatesResult.StatesFetched
+import org.simple.clinic.util.ResolvedError
+import retrofit2.HttpException
```

Comment:
Having retrofit imports in a test for the business logic seems like a code smell to me. I feel we should move the conversion of http errors to the screen level errors in the effect handler so that the update function does not know about anything outside its domain. What do you think?

Reply:
Ah, that explains why we are doing it like that in the manifest/country selection screen. I will move them.